packages/cozy-harvest-lib/src/components/cards/LaunchTriggerCard.jsx

```diff
@@ -10,83 +10,76 @@ import Uppercase, { Text } from 'cozy-ui/transpiled/react/Text'
-    return (
-      <Card className={className}>
-        <FlowProvider
-          {...pick(this.props, Object.keys(DumbFlowProvider.propTypes))}
-        >
-          {({ flow }) => {
-            const flowState = flow.getState()
-            const launch = flow.launch
-            const trigger = flow.trigger
-            const running = flowState.running
-            const lastSuccessDate = triggers.getLastSuccessDate(trigger)
-            return (
-              <div className="u-flex u-flex-column-s">
-                <ul className="u-nolist u-m-0 u-mr-1 u-pl-0 u-flex-grow-1">
-                  <li className="u-mb-half">
-                    <Uppercase
-                      tag="span"
-                      className="u-coolGrey u-mr-half u-fz-tiny"
-                    >
-                      {t('card.launchTrigger.lastSync.label')}
-                    </Uppercase>
-                    <Text className="u-fz-tiny" tag="span">
-                      {running
-                        ? t('card.launchTrigger.lastSync.syncing')
-                        : lastSuccessDate
-                        ? f(
-                            lastSuccessDate,
-                            t('card.launchTrigger.lastSync.format')
-                          )
-                        : t('card.launchTrigger.lastSync.unknown')}
-                    </Text>
-                  </li>
-                  <li className="u-mb-half">
-                    <Uppercase
-                      className="u-coolGrey u-mr-half u-fz-tiny"
-                      tag="span"
-                    >
-                      {t('card.launchTrigger.frequency.label')}
-                    </Uppercase>
-                    <Text className="u-fz-tiny" tag="span">
-                      {t(
-                        `card.launchTrigger.frequency.${triggers.getFrequency(
-                          trigger
-                        ) || 'undefined'}`
-                      )}
-                    </Text>
-                  </li>
-                </ul>
-                <div>
-                  <Button
-                    label={t('card.launchTrigger.button.label')}
-                    icon={<Icon focusable="false" icon="sync" spin={running} />}
-                    className="u-mh-0 u-mv-0"
-                    disabled={running || disabled}
-                    onClick={() => launch(trigger)}
-                    subtle
-                    // TODO: Extract this directly in Cozy-UI
-                    // (either with an utility class or a Button prop)
-                    style={{ lineHeight: '1.4' }}
-                  />
-                </div>
-              </div>
-            )
-          }}
-        </FlowProvider>
-      </Card>
-    )
+const LaunchTriggerCard = props => {
+  if (props.flow) {
+    return <DumbLaunchTriggerCard {...props} flow={props.flow} />
   }
+  return (
+    <FlowProvider {...pick(props, Object.keys(DumbFlowProvider.propTypes))}>
```

Comment:
How do you feel about changing this `pick()` call? I believe it's the one that caused problems when we removed the proptypes from the built packages, and it also makes it hard to understand what props are being passed to `FlowProvider` without looking at it's proptypes.

Reply:
I'd be happy to remove it, it's a smell.

Reply:
I [changed the API to use flowProps](https://github.com/cozy/cozy-libs/pull/984/commits/cef9dba413d0f99346a9835a08e13e77df835779) instead of directly spreading with pick + FlowProvider.propTypes. Backward compatibility is conserved with props.initialTrigger.